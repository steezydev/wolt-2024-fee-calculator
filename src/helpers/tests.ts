const localStorageMock = (function () {
    let store: Record<string, string> = {};
    return {
        getItem(key: string) {
            return store[key] || null;
        },
        setItem(key: string, value: string) {
            store[key] = value.toString();
        },
        clear() {
            store = {};
        },
    };
})();

const matchMediaMock = jest.fn().mockImplementation((query) => ({
    matches: query === '(prefers-color-scheme: dark)',
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
}));

export const mockContext = () => {
    beforeAll(() => {
        Object.defineProperty(window, 'localStorage', {
            value: {
                getItem: jest.fn().mockImplementation((key: string) => {
                    return localStorageMock.getItem(key);
                }),
                setItem: jest
                    .fn()
                    .mockImplementation((key: string, value: string) => {
                        localStorageMock.setItem(key, value);
                    }),
                clear: localStorageMock.clear,
            },
            writable: true,
        });

        window.matchMedia = matchMediaMock;
    });

    beforeEach(() => {
        (window.localStorage.getItem as jest.Mock).mockClear();
        (window.localStorage.setItem as jest.Mock).mockClear();
        matchMediaMock.mockClear();
    });
};

export const setupFakeDate = () => {
    beforeEach(() => {
        jest.useFakeTimers();
        jest.setSystemTime(new Date('2024-01-01'));
    });

    afterEach(() => {
        jest.useRealTimers();
    });
};

export const mockScrollTo = () => {
    beforeEach(() => {
        Element.prototype.scrollTo = jest.fn();
    });

    afterEach(() => {
        (Element.prototype.scrollTo as jest.Mock).mockRestore();
    });
};
