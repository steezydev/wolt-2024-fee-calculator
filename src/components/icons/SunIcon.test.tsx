import { iconRenderTests } from './Icon.tests';
import SunIcon from './SunIcon';

iconRenderTests('SunIcon', SunIcon);
