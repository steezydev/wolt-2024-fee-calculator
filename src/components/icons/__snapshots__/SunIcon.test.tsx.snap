// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SunIcon when asked to render matches snapshot 1`] = `
<svg
  aria-hidden="true"
  fill="none"
  height="24"
  stroke="currentColor"
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth="2"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="12"
    cy="12"
    r="4"
  />
  <path
    d="M12 2v2"
  />
  <path
    d="M12 20v2"
  />
  <path
    d="m4.93 4.93 1.41 1.41"
  />
  <path
    d="m17.66 17.66 1.41 1.41"
  />
  <path
    d="M2 12h2"
  />
  <path
    d="M20 12h2"
  />
  <path
    d="m6.34 17.66-1.41 1.41"
  />
  <path
    d="m19.07 4.93-1.41 1.41"
  />
</svg>
`;

exports[`SunIcon when asked to render with additional class renders with additional class 1`] = `
<svg
  aria-hidden="true"
  className="some-class"
  fill="none"
  height="24"
  stroke="currentColor"
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth="2"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="12"
    cy="12"
    r="4"
  />
  <path
    d="M12 2v2"
  />
  <path
    d="M12 20v2"
  />
  <path
    d="m4.93 4.93 1.41 1.41"
  />
  <path
    d="m17.66 17.66 1.41 1.41"
  />
  <path
    d="M2 12h2"
  />
  <path
    d="M20 12h2"
  />
  <path
    d="m6.34 17.66-1.41 1.41"
  />
  <path
    d="m19.07 4.93-1.41 1.41"
  />
</svg>
`;
